src/Moq/ISetup.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2007, Clarius Consulting, Manas Technology Solutions, InSTEDD.
+// All rights reserved. Licensed under the BSD 3-Clause License; see License.txt.
+
+using System.Linq.Expressions;
+
+namespace Moq
+{
+	/// <summary>
+	///   A setup configured on a mock.
+	/// </summary>
+	public interface ISetup
+	{
+		/// <summary>
+		///   The setup expression.
+		/// </summary>
+		LambdaExpression Expression { get; }
+
+		/// <summary>
+		///   Gets whether this setup is conditional.
+		/// </summary>
+		bool IsConditional { get; }
+
+		/// <summary>
+		///   Gets whether this setup was matched by at least one invocation on the mock.
+		/// </summary>
+		bool IsMatched { get; }
```

Comment:
Just to clarify, if this is `false` and I want to convert it into a nice exception in a specific place, I can call `Verify()`?

Reply:
`Verify` will *typically* throw if `IsMatched` is `false`&mdash;but there are some internal setup types where that isn't necessarily the case (among other reasons, for backward compatibility).

For instance, if a `get`/`set` property gets setup by `SetupAllProperties`, two setups are created: one for the getter, one for the setter. Both of these setups will verify successfully even if they were never matched by an invocation.

TL;DR: Even if `IsMatched` is `false`, `Verify` may succeed and give you no error message. If you want to verify, call `Verify`. :-)

I'll eventually try to document the exact mode of operation of `Verify` et al., but I'm not quite there yet.